client/src/components/ModelViewerComponent/ModelViewerComponent.test.js

```diff
@@ -0,0 +1,318 @@
+  apiService.queryModels.mockResolvedValue(models);
+  ModelService.prototype.deleteAll.mockResolvedValue(mockSuccessResponse);
+  act(() => {
+    render(<ModelViewerComponent showItemMenu="true" />, container);
+  });
+
+  await findByText(container, "Floor");
+  const button = await findByLabelText(container, "modelViewerCommandBarComponent.farItems.deleteModels.ariaLabel");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  await findByText(container, "Are you sure?");
+  const deleteButton = container.querySelector(".save-button");
+  act(() => {
+    deleteButton.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  expect(ModelService.prototype.deleteAll.mock.calls.length).toBe(1);
+});
+
+test("upload model", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.addModels.mockResolvedValue(mockSuccessResponse);
+  ModelService.prototype.getModelIdsForUpload.mockResolvedValue([ "dtmi:com:example:adtexplorer:FloorNew;1" ]);
+  ModelService.prototype.chunkModelsList.mockReturnValue([ [ "dtmi:com:example:adtexplorer:FloorNew;1" ] ]);
+  act(() => {
+    render(<ModelViewerComponent showItemMenu="true" />, container);
+  });
+
+  await findByText(container, "Floor");
+  const button = await findByLabelText(container, "modelViewerCommandBarComponent.farItems.uploadModel.text");
+  expect(uploadModel).toHaveBeenCalledTimes(0);
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  const str = JSON.stringify(uploadValue);
+  const blob = new Blob([ str ]);
+  const file = new File([ blob ], "values.json", {
+    type: "application/json"
+  });
+  File.prototype.text = jest.fn().mockResolvedValueOnce(str);
+  const input = container.querySelectorAll(".mv-fileInput");
+  userEvent.upload(input[0], file);
+  await new Promise(resolve => setTimeout(resolve, 100));
+  expect(uploadModel).toHaveBeenCalledTimes(1);
+});
+
+
+test("upload model image", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  settingsService.setModelImage.mockResolvedValue(mockSuccessResponse);
+  act(() => {
+    render(<ModelViewerComponent showItemMenu="true" />, container);
+  });
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.uploadModel");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  const blob = new Blob([ mockGetModelImage ]);
+  const file = new File([ blob ], "Floor.json", {
+    type: "application/JSON"
+  });
```

Comment:
Should we update the file to match the content? I.e. an image rather than a json file.